LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceRegistry.cs

```diff
@@ -215,6 +215,15 @@ public async Task<LoRaDevice> GetDeviceForJoinRequestAsync(string devEUI, DevNon
 
             if (searchDeviceResult.IsDevNonceAlreadyUsed)
             {
+                // another gateway processed the join request. If we have it in the cache
+                // with existing session keys, we need to invalidate that entry, to ensure
+                // it gets re-fetched on the next message
+                if (this.deviceCache.TryGetByDevEui(devEUI, out var someDevice) && someDevice.AppSKey != null)
```

Comment:
I don't understand the case `someDevice.AppSKey != null`, why would we have it in the cache without an `AppSKey`?

Reply:
As discussed: when a join request is received, we do put it into the cache without the session keys yet initialized. That means, we can't handle incoming messages, but we can use it for other join messages (if we miss the join receive window).